src/settings/SettingsStore.js

```diff
@@ -260,7 +260,7 @@ export default class SettingsStore {
      * @param {*} value The new value of the setting, may be null.
      * @return {Promise} Resolves when the setting has been changed.
      */
-    static setValue(settingName, roomId, level, value) {
+    static async setValue(settingName, roomId, level, value) {
```

Comment:
I'm very worried about making this async as I'm fairly confident that there are places in the project that expect this to be sync.

Reply:
But it already returned a promise before? The external API hasn't changed AFAIK.

Reply:
E.g. this is just about readability AFAIAC. The behaviour shouldn't have changed at all now canChangeTo is removed. I can still take it out if you prefer.

Reply:
Ah, I see. It should be the same, yes. Mixing up my mental iterations of this section